beeai/supervisor/http_utils.py

```diff
@@ -0,0 +1,121 @@
+from contextlib import asynccontextmanager
+import aiohttp
+import os
+import ssl
+
+import requests
+import requests.adapters
+
+_aiohttp_session: aiohttp.ClientSession | None = None
```

Comment:
should this (and the requests one below) be ContextVars ?

Reply:
it doesn't matter for the usage, but yes! I thought "no thread usage, I don't need threading.local()" without considering that this is an async context manager that can yield.

I then found out that the REDHAT_IT_CA_BUNDLE handling here is unnecessary, and we can just bake it into the image so the utility of all this is reduced, but I think I'll keep it:
 - To get proper session reuse
 - As a placeholder for future session setup with timeouts / retries.